src/AppInstallerCLICore/ConfigurationSetProcessorFactoryRemoting.cpp

```diff
@@ -298,9 +331,54 @@ namespace AppInstaller::CLI::ConfigurationRemoting
         };
     }
 
-    IConfigurationSetProcessorFactory CreateOutOfProcessFactory(bool useRunAs, const std::string& properties, const std::string& restrictions)
+    IConfigurationSetProcessorFactory CreateOutOfProcessFactory(ProcessorEngine processorEngine, bool useRunAs, const std::string& properties, const std::string& restrictions)
+    {
+        THROW_HR_IF(APPINSTALLER_CLI_ERROR_EXPERIMENTAL_FEATURE_DISABLED, processorEngine == ProcessorEngine::DSCv3 && !Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::ConfigurationDSCv3));
+
+        return winrt::make<RemoteFactory>(processorEngine, useRunAs, properties, restrictions);
+    }
+
+    ProcessorEngine DetermineProcessorEngine(ConfigurationSet set)
     {
-        return winrt::make<RemoteFactory>(useRunAs, properties, restrictions);
+        Utility::Version schemaVersion{ Utility::ConvertToUTF8(set.SchemaVersion()) };
+
+        if (schemaVersion <= Utility::Version{ "0.3" })
+        {
+            ProcessorEngine result = ProcessorEngine::Unknown;
+
+            winrt::hstring processorIdentifier = set.Environment().ProcessorIdentifier();
+            if (processorIdentifier.empty() || processorIdentifier == L"pwsh")

```

Comment:
Do we want to be more permissive by doing case insensitive checks?

Reply:
nit: btw, the tests in ConfigurationSetAuthoringTests.cs used dsc3 as ProcessorIdentifier,  we may want to update those to avoid confusion.